### PR TITLE
Refactor of Internal/PKG

### PR DESCRIPTION
**Pull request by da-ar**
Sorry about the size of this PR it took a bit of fettling to get this in shape. 
There are tests commented out that are no longer directly testable. And it can be seen that the PCT api is not fully tested.

### CODE REVIEW

File: internal/pkg/pct/pct_test.go [hunk 1]
```diff
@@ -1,4 +1,4 @@
-package pct
+package pct_test
```
Comment (sanfrancrisko) [inline internal/pkg/pct/pct_test.go:1]: Just capturing the comment I made on the call about a "phase 2" to move some of the mock file declarations out in to separate files. A tidy up of the duplicate test resources too would be good - again not a comment for you to action on this PR, but just wanting to capture this.

File: internal/pkg/pct/pct_test.go [hunk 2]
```diff
@@ -172,7 +221,31 @@ func TestDeploy(t *testing.T) {
...
 		t.Run(tt.name, func(t *testing.T) {
-			if got := Deploy(tt.args.info); !reflect.DeepEqual(got, tt.want) {
+			fs := afero.NewMemMapFs()
+			afs := &afero.Afero{Fs: fs}
+			iofs := &afero.IOFS{Fs: fs}
+
+			// Create the template
+			templateDir := filepath.Join(tt.args.info.TemplateCache, tt.args.info.SelectedTemplate)
+			contentDir := filepath.Join(templateDir, "content")
+			afs.MkdirAll(contentDir, 755) //nolint:errcheck
```
Comment (sanfrancrisko) [inline internal/pkg/pct/pct_test.go:231]: Similar to my earlier comment:
```suggestion
			afs.MkdirAll(contentDir, 750)
```

File: internal/pkg/pct/pct_test.go [hunk 3]
```diff
@@ -213,304 +305,39 @@ func TestGet(t *testing.T) {
...
-			if err := createTemplateFile(tt.args.info, tt.args.configFile, tt.args.templateFile, tt.args.tmpl); (err != nil) != tt.wantErr {
-				t.Errorf("createTemplateFile() error = %v, wantErr %v", err, tt.wantErr)
-			}
-			if _, err := os.Stat(tt.args.templateFile.TargetFilePath); err != nil {
-				t.Errorf("createTemplateFile() error = %v, wantErr %v", err, tt.wantErr)
+			if tt.args.setup {
+				// Create the template
+				templateDir := filepath.Join(tt.args.templateCache, tt.args.selectedTemplate)
+				contentDir := filepath.Join(templateDir, "content")
+				afs.MkdirAll(contentDir, 755) //nolint:errcheck
```
Comment (sanfrancrisko) [inline internal/pkg/pct/pct_test.go:316]: Similar to my earlier comment:
```suggestion
				afs.MkdirAll(contentDir, 750)
```

File: internal/pkg/utils/filepath_utils.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
+package utils
```
Comment (sanfrancrisko) [inline internal/pkg/utils/filepath_utils.go:1]: Can we remove this file if it's not being used anywhere?

File: internal/pkg/utils/io_utils.go [hunk 5]
```diff
@@ -2,12 +2,12 @@ package utils
...
+type IoUtilI interface {
 	TempDir() (name string, err error)
 }
 
-type IoUtilHelpersImpl struct{}
+type IoUtil struct{}
 
-func (IoUtilHelpersImpl) TempDir() (name string, err error) {
+func (IoUtil) TempDir() (name string, err error) {
 	return ioutil.TempDir("", "")
```
Comment (sanfrancrisko) [inline internal/pkg/utils/io_utils.go:12]: Is this going to remain as a call to `ioutil.TempDir` or should we move to Afero's implementation for consistency?

**Review by sanfrancrisko — COMMENTED** [review]
Excellent job! 👍 

Some minor changes, a few queries and some comments in to capture stuff in our conversations over the past few days.

File: internal/pkg/gzip/gzip_test.go [hunk 6]
```diff
@@ -1,13 +1,18 @@
...
+	"github.com/puppetlabs/pdkgo/internal/pkg/gzip"
+	"github.com/spf13/afero"
 )
 
 func TestGzip(t *testing.T) {
-	gzipOutputPath := t.TempDir()
+	fs := afero.NewMemMapFs()
+	afs := &afero.Afero{Fs: fs}
+
+	gzipOutputPath, _ := afs.TempDir("", "")
```
[the remark this answers is not in the document]
Comment (da-ar) [reply internal/pkg/gzip/gzip_test.go:15]: I'll need to see a reproduction of that. As this is using MemMapFs for the filesystem I would doubt that it is creating a temp directory with the Mac symlink issue.

**Review by sanfrancrisko — APPROVED** [review]

**Pull request merged**